### PR TITLE
MET-4614 ZipFileReader CompressedFileExtractor unification

### PR DESCRIPTION
**Pull request by adolfopeixinho**

### CODE REVIEW

File: metis-repository/src/test/java/eu/europeana/metis/repository/rest/RecordControllerTest.java [hunk 1]
```diff
@@ -192,7 +192,9 @@ void saveRecords_Exception() throws Exception {
                             .param("dateStamp", "+1000000000-12-31T23:59:59.999999999Z")
                             .contentType(MediaType.MULTIPART_FORM_DATA_VALUE))
                         .andDo(print())
-                        .andExpect(status().is(500));
+                        .andExpect(status().is(500))
+                        .andExpect(result -> assertTrue(result.getResolvedException() instanceof ResponseStatusException));
+
```
Comment (jeortizquan) [inline metis-repository/src/test/java/eu/europeana/metis/repository/rest/RecordControllerTest.java:197]: Here is a suggestion to make more readable the assertion.

```
 @Test
  void saveRecords_Exception() throws Exception {
    InputStream inputStream = this.getClass().getClassLoader().getResourceAsStream("repository-test-error.zip");
    MockMultipartFile recordsFile = new MockMultipartFile("recordsZipFile",
        "repository-test-error.zip",
        "application/zip",
        inputStream);
    when(recordDaoMock.createRecord(any(Record.class))).thenReturn(true);
    recordController.setRecordDao(recordDaoMock);
    MvcResult result = recordControllerMock.perform(multipart(RestEndpoints.REPOSITORY_RECORDS)
                            .file(recordsFile)
                            .param("datasetId", "datasetId")
                            .param("dateStamp", "+1000000000-12-31T23:59:59.999999999Z")
                            .contentType(MediaType.MULTIPART_FORM_DATA_VALUE))
                        .andDo(print())
                        .andExpect(status().is(500))
                        .andReturn();
    assertTrue(result.getResolvedException() instanceof ResponseStatusException);
    verify(recordDaoMock, times(0)).createRecord(any());
  }
```

File: metis-validation/metis-validation-rest/src/main/java/eu/europeana/validation/rest/ValidationController.java [hunk 2]
```diff
@@ -87,8 +87,8 @@ public ValidationResult validate(
   }
 
   /**
```
Comment (jeortizquan) [inline metis-validation/metis-validation-rest/src/main/java/eu/europeana/validation/rest/ValidationController.java:89]: Fix code smell in line 74 for variable ```record``` a rename could be ```recordToValidate```
also remember to update the javadoc.

```
 @PostMapping(value = RestEndpoints.SCHEMA_VALIDATE, consumes = APPLICATION_XML_VALUE, produces = APPLICATION_JSON_VALUE)
  @ResponseBody
  @ApiOperation(value = "Validate single record based on schema", response = ValidationResult.class)
  public ValidationResult validate(
      @ApiParam(value = "schema") @PathVariable("schema") String targetSchema,
      @RequestBody String recordToValidate
  ) throws ValidationException {
    if (!schemaProvider.isPredefined(targetSchema)) {
      throw new ValidationException("", "", "It is not predefined schema.");
    }

    ValidationResult result = validator.singleValidation(targetSchema, null, null, recordToValidate);
    if (result.isSuccess()) {
      return result;
    } else {
      LOGGER.error(result.getMessage());
      throw new ValidationException(result.getRecordId(), result.getNodeId(), result.getMessage());
    }
  }
```

File: metis-validation/metis-validation-rest/src/main/java/eu/europeana/validation/rest/ValidationController.java [hunk 3]
```diff
@@ -112,28 +112,28 @@ public ValidationResultList batchValidate(
...
+        if (validationResultList.getResultList() != null || validationResultList.getResultList().isEmpty()) {
+          validationResultList.setSuccess(true);
+          return validationResultList;
+        }
+        throw new BatchValidationException("Batch validation service failed.", validationResultList);
       }
-    } catch (ExecutionException e) {
-      throw new ServerException(e);
-    } catch (InterruptedException e) {
+      return null;
```
Comment (jeortizquan) [inline metis-validation/metis-validation-rest/src/main/java/eu/europeana/validation/rest/ValidationController.java:134]: I think this ```return null``` goes at the end. what do you think?
```
@PostMapping(value = SCHEMA_BATCH_VALIDATE)
  @ResponseBody
  @ApiOperation(value = "Validate zip file based on schema", response = ValidationResultList.class)
  public ValidationResultList batchValidate(
      @ApiParam(value = "schema") @PathVariable("schema") String targetSchema,
      @ApiParam(value = "file") @RequestPart("file") MultipartFile providedZipFile)
      throws ServerException, BatchValidationException {

    if (!schemaProvider.isPredefined(targetSchema)) {
      throw new BatchValidationException("It is not predefined schema.",
          new ValidationResultList());
    }

    final List<ByteArrayInputStream> records;
    try {
      records = new CompressedFileHandler().getContentFromZipFile(providedZipFile.getInputStream());
    } catch (IOException e) {
      throw new ServerException(e);
    }
    if (records.isEmpty()) {
      throw new ServerException("No suitable records found in zip file.");
    }

    try {
      ValidationResultList validationResultList =
          validator.batchValidation(targetSchema, null, null, records);

      if (validationResultList != null) {
        if (validationResultList.getResultList() != null || validationResultList.getResultList().isEmpty()) {
          validationResultList.setSuccess(true);
          return validationResultList;
        }
        throw new BatchValidationException("Batch validation service failed.", validationResultList);
      }
    } catch (ExecutionException | InterruptedException e) {
      Thread.currentThread().interrupt();
      throw new ServerException(e);
    }
    return null;
  }
```

File: pom.xml [hunk 4]
```diff
@@ -136,7 +136,7 @@
     <version.com.github.tomakehurst.wiremock>2.27.2</version.com.github.tomakehurst.wiremock>
     <version.com.zoho.crm.java-sdk>3.0.1</version.com.zoho.crm.java-sdk>
     <version.apache.commons.text>1.9</version.apache.commons.text>
-    <version.commonscompress>1.11</version.commonscompress>
+    <version.commonscompress>1.21</version.commonscompress>
```
Comment (jeortizquan) [inline pom.xml:139]: :+1:

**Review by jeortizquan — CHANGES_REQUESTED** [review]
I think the code changes include good improvements, just have a look on the comments.

File: metis-validation/metis-validation-rest/src/main/java/eu/europeana/validation/rest/ValidationController.java (hunk 3 above)
Comment (adolfopeixinho) [reply metis-validation/metis-validation-rest/src/main/java/eu/europeana/validation/rest/ValidationController.java:134]: Of course it goes ;)

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**